### ISSUE TITLE
Movement Fix

### ISSUE THREAD

**Issue by Meeplez** [NecronomiconCoding/Pokemon-Go-Rocket-API#58]
Movement as of Current Teleports, This means Wider ban risk, and no egg farming :/
There has been some stuff posted on other forks,
I dont take credit, however you should write your own,

` private async Task<PlayerUpdateResponse> DoWalk()
        {
            System.Console.ForegroundColor = ConsoleColor.Cyan;
            Console.WriteLine($"Do walk to lat: {_currentLat} lng: {_currentLng}");
            System.Console.ResetColor();
            var customRequest = new Request.Types.PlayerUpdateProto
            {
                Lat = Utils.FloatAsUlong(_currentLat),
                Lng = Utils.FloatAsUlong(_currentLng)
            };

            var updateRequest = RequestBuilder.GetRequest(_unknownAuth, _currentLat, _currentLng, 10,
                new Request.Types.Requests
                {
                    Type = (int)RequestType.PLAYER_UPDATE,
                    Message = customRequest.ToByteString()
                });
            var updateResponse =
                await _httpClient.PostProtoPayload<Request, PlayerUpdateResponse>($"https://{_apiUrl}/rpc", updateRequest);

            return updateResponse;
        }

        public async Task<PlayerUpdateResponse> UpdatePlayerLocation(double lat, double lng)
        {
            PlayerUpdateResponse updateResponse = null;

            if (_currentLat >= lat)
            {
                while (_currentLat > lat)
                {
                    SetCoordinates(_currentLat -= 0.000095, _currentLng);
                    updateResponse = await DoWalk();
                    await Task.Delay(100);
                }
            }
            else
            {
                while (_currentLat < lat)
                {
                    SetCoordinates(_currentLat += 0.000095, _currentLng);
                    updateResponse = await DoWalk();
                    await Task.Delay(100);
                }
            }

            if (_currentLng >= lng)
            {
                while (_currentLng > lng)
                {
                    SetCoordinates(_currentLat, _currentLng -= 0.000095);
                    updateResponse = await DoWalk();
                    await Task.Delay(100);
                }
            }
            else
            {
                while (_currentLng < lng)
                {
                    SetCoordinates(_currentLat, _currentLng += 0.000095);
                    updateResponse = await DoWalk();
                    await Task.Delay(100);
                }
            }

            return updateResponse;
        }`

**Comment by Krexis** [comment]
You sure it teleports? Cause my eggs get hatched lol.

**Comment by Meeplez** [comment]
yeah  was running this last night and only sometimes " i think" eggsteps++
